kernel/src/engine/arrow_expression/in_list.rs

```diff
@@ -0,0 +1,566 @@
+                            "Cannot cast {} to {}", column.data_type(), data_type
+                        )))
+                };
+            }
+
+            // safety: as_* methods on arrow arrays can panic, but we checked the data type before applying.
+            match (column.data_type(), data_type) {
+                (ArrowDataType::Utf8, PrimitiveType::String) => {
+                    is_in_list(ad, scalars_from(column_as!(string::<i32>)?))
+                }
+                (ArrowDataType::LargeUtf8, PrimitiveType::String) => {
+                    is_in_list(ad, scalars_from(column_as!(string::<i64>)?))
+                }
+                (ArrowDataType::Utf8View, PrimitiveType::String) => {
+                    is_in_list(ad, scalars_from(column_as!(string_view)?))
+                }
+                (ArrowDataType::Boolean, PrimitiveType::Boolean) => {
+                    is_in_list(ad, scalars_from(column_as!(boolean)?))
+                }
+                (ArrowDataType::Binary, PrimitiveType::Binary) => {
+                    is_in_list(ad, scalars_from(column_as!(binary::<i32>)?))
+                }
+                (ArrowDataType::LargeBinary, PrimitiveType::Binary) => {
+                    is_in_list(ad, scalars_from(column_as!(binary::<i64>)?))
+                }
+                (ArrowDataType::BinaryView, PrimitiveType::Binary) => {
+                    is_in_list(ad, scalars_from(column_as!(binary_view)?))
+                }
+                (_, PrimitiveType::Byte) => {
+                    is_in_list(ad, scalars_from(column_as!(primitive::<Int8Type>)?))
+                }
+                (_, PrimitiveType::Short) => {
+                    is_in_list(ad, scalars_from(column_as!(primitive::<Int16Type>)?))
+                }
+                (_, PrimitiveType::Integer) => {
+                    is_in_list(ad, scalars_from(column_as!(primitive::<Int32Type>)?))
+                }
+                (_, PrimitiveType::Long) => {
+                    is_in_list(ad, scalars_from(column_as!(primitive::<Int64Type>)?))
+                }
+                (_, PrimitiveType::Float) => {
+                    is_in_list(ad, scalars_from(column_as!(primitive::<Float32Type>)?))
+                }
+                (_, PrimitiveType::Double) => {
+                    is_in_list(ad, scalars_from(column_as!(primitive::<Float64Type>)?))
+                }
+                (_, PrimitiveType::Date) => is_in_list(
+                    ad,
+                    to_scalars(column_as!(primitive::<Date32Type>)?, Scalar::Date),
+                ),
+                (
+                    ArrowDataType::Timestamp(TimeUnit::Microsecond, Some(_)),
+                    PrimitiveType::Timestamp,
+                ) => is_in_list(
+                    ad,
+                    to_scalars(
+                        column_as!(primitive::<TimestampMicrosecondType>)?,
+                        Scalar::Timestamp,
+                    ),
+                ),
```

Comment:
nit: not sure it's worth the trouble, but could do
```suggestion
                ) => {
                    let column = column_as!(primitive::<TimestampMicrosecondType>)?;
                    is_in_list(ad, to_scalars(column, Scalar::Timestamp))
                }
```
(again below)